interview-questions.md

```diff
@@ -8,48 +8,72 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-Your answer:
+Your answer: Yarn is a package manager that we use as developers to access the collaborative code of other developers, so that we can create smoother applications, faster. I've previously used yarn for jest testing, but I've also used it to compile the files necessary for React apps. When you run yarn to create a new React app, it will add a folder called node_modules which contains all of the source code that allows the application to function, as well as a public and src folder that contain various components like your index.html file and App.js.
 
-Researched answer:
+Researched answer: In addition to my previous answer, yarn stands for "Yet Another Resource Negotiator", and was created by Facebook in 2016 for Node.js.
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: An anonymous function in Javascript is, as the name implies, a function that has no name. One of the places you'll see them commonly used is in methods, which will work on specific data types like objects and arrays. For example, in the array method .map(), you will pass in an anonymous function that will accept up to three parameters. (array.map((value, index, origArr) => )).
 
-Researched answer:
+Researched answer: In addition to my previous answer, it's important to note that anonymous functions CANNOT be called after their creation, as there would be no name to call it by. They can also accept multiple arguments, but can only have one expression.
 
 4. What is the difference between state values and props in React?
 
-Your answer:
+Your answer: In react, state values are values that can be updated in real-time via the Virtual DOM without refreshing the full page. The format for that looks like this: - Declaring state: [value, methodToSetValue] = tellsReactToUseState(initialValue)
+const [state, setState] = useState(initialValue)
+They're great for storing data that we will want to update later!
+Props come in handy when you have multiple components (isolated code stored on another file which will be connected and rendered on your main page) that you want to give access to that state. We do this through the use of props (short for properties), which we can then pass down to a child component like so:
+<Component dataName="data"/>
+That data can then be accessed from the child component, like so:
+const Component = (props) => {
+
+<div>{props.dataName}</div>
+}
 
-Researched answer:
+Researched answer: In addition to my previous answer, it's important to note that states and props are both JS objects. This means that to access data that's being stored inside of them, we need to remember to either use dot notation, or destructure that data at the top of our code to be accessed later.
```

Comment:
While props is a special JS object, state is simply a variable that can be any valid data type.